File: para-claude-code/update-html.js
const fs = require('fs');
const path = require('path');

const htmlFile = path.join(__dirname, 'Assistente-Onde-Morar-Portugal.html');

// Read the HTML file
let htmlContent = fs.readFileSync(htmlFile, 'utf-8');
console.log(`📍 Starting HTML updates...`);
console.log(`   Original HTML size: ${htmlContent.length} characters`);

// Read the schools array
const schoolsContent = fs.readFileSync(path.join(__dirname, 'schools-for-html.txt'), 'utf-8').trim();

// Read the concelhos array
const concelhosContent = fs.readFileSync(path.join(__dirname, 'concelhos-for-html.txt'), 'utf-8').trim();

// ===== REPLACE SCHOOLS ARRAY =====
// Find and replace schools array (const schools = [...];)
const schoolsRegex = /const schools = \[[\s\S]*?\];/;
const schoolsMatch = htmlContent.match(schoolsRegex);
if (schoolsMatch) {
  const oldSchools = schoolsMatch[0];
  htmlContent = htmlContent.replace(schoolsRegex, schoolsContent);
  console.log(`✓ Replaced schools array`);
  console.log(`  Old: ${oldSchools.length} chars, New: ${schoolsContent.length} chars`);
} else {
  console.error('✗ Could not find schools array!');
  process.exit(1);
}

// ===== REPLACE PLACES ARRAY =====
// Find and replace places array (const places = [...];)
const placesRegex = /const places = \[[\s\S]*?\n\];/;
const placesMatch = htmlContent.match(placesRegex);
if (placesMatch) {
  const oldPlaces = placesMatch[0];
  htmlContent = htmlContent.replace(placesRegex, concelhosContent);
  console.log(`✓ Replaced places array with concelhos`);
  console.log(`  Old: ${oldPlaces.length} chars, New: ${concelhosContent.length} chars`);
} else {
  console.error('✗ Could not find places array!');
  process.exit(1);
}

// ===== UPDATE COMMENT =====
// Update the data load comment
const oldComment = `// Use CONCELHOS from concelhos.js if available, otherwise fall back to places array\nvar placesData = window.CONCELHOS || places;`;
const newComment = `// Using updated 94-municipality concelhos array (from concelhos-corrigido.js)\nvar placesData = window.CONCELHOS || concelhos;`;
htmlContent = htmlContent.replace(oldComment, newComment);
console.log(`✓ Updated data load comment`);

// Write the updated HTML
fs.writeFileSync(htmlFile, htmlContent, 'utf-8');

console.log(`\n✅ HTML updated successfully!`);
console.log(`   New HTML size: ${htmlContent.length} characters`);
console.log(`\n📊 Summary:`);
console.log(`   - Schools: 62 entries (Aljezur removed)`);
console.log(`   - Concelhos: 94 municipalities`);
console.log(`   - BGA hubs: 24 included as separate entries`);

// Validation
if (htmlContent.includes('var concelhos = [')) {
  console.log(`✓ Concelhos array correctly inserted`);
}
if (htmlContent.includes('const schools = [')) {
  console.log(`✓ Schools array correctly inserted`);
}
if (htmlContent.includes('praiasFluviais')) {
  console.log(`✓ River beaches field present`);
}
if (htmlContent.includes('Monchique')) {
  console.log(`✓ Monchique spelling correct (not Monchaique)`);
}

console.log(`\n⚠️  Next steps:`);
console.log(`   1. Test in browser - DevTools Console: concelhos.length (should be 94)`);
console.log(`   2. Test: schools.length (should be 62)`);
console.log(`   3. Test filters: 'non-hot + car + river beaches' → interior areas`);
console.log(`   4. Verify Aljezur not in schools array`);
console.log(`   5. Test school proximity scoring`);

File: para-claude-code_backup/update-html.py
#!/usr/bin/env python3

import re

# Read the HTML file
html_file = 'Assistente-Onde-Morar-Portugal.html'
with open(html_file, 'r', encoding='utf-8') as f:
    html_content = f.read()

# Read the schools array
with open('schools-for-html.txt', 'r', encoding='utf-8') as f:
    schools_content = f.read().strip()

# Read the concelhos array
with open('concelhos-for-html.txt', 'r', encoding='utf-8') as f:
    concelhos_content = f.read().strip()

print("📍 Starting HTML updates...")
print(f"   Original HTML size: {len(html_content)} characters")

# ===== REPLACE SCHOOLS ARRAY =====
# Find the schools array (const schools = [...];)
schools_pattern = r'const schools = \[.*?\];'
schools_match = re.search(schools_pattern, html_content, re.DOTALL)
if schools_match:
    old_schools = schools_match.group(0)
    html_content = html_content.replace(old_schools, schools_content)
    print(f"✓ Replaced schools array")
    print(f"  Old: {len(old_schools)} chars, New: {len(schools_content)} chars")
else:
    print("✗ Could not find schools array!")
    exit(1)

# ===== REPLACE PLACES ARRAY =====
# Find the places array (const places = [...];)
places_pattern = r'const places = \[.*?\n\];'
places_match = re.search(places_pattern, html_content, re.DOTALL)
if places_match:
    old_places = places_match.group(0)
    html_content = html_content.replace(old_places, concelhos_content)
    print(f"✓ Replaced places array with concelhos")
    print(f"  Old: {len(old_places)} chars, New: {len(concelhos_content)} chars")
else:
    print("✗ Could not find places array!")
    exit(1)

# ===== UPDATE COMMENT =====
# Update the data load comment to reflect new data source
html_content = html_content.replace(
    "// Use CONCELHOS from concelhos.js if available, otherwise fall back to places array\nvar placesData = window.CONCELHOS || places;",
    "// Using updated 94-municipality concelhos array (from concelhos-corrigido.js)\nvar placesData = window.CONCELHOS || concelhos;"
)
print("✓ Updated data load comment")

# Write the updated HTML
with open(html_file, 'w', encoding='utf-8') as f:
    f.write(html_content)

print(f"\n✅ HTML updated successfully!")
print(f"   New HTML size: {len(html_content)} characters")
print(f"\n📊 Summary:")
print(f"   - Schools: 62 entries (Aljezur removed)")
print(f"   - Concelhos: 94 municipalities")
print(f"   - BGA hubs: 24 included as separate entries")

# Validation
if 'var concelhos = [' in html_content:
    print("✓ Concelhos array correctly inserted")
if 'const schools = [' in html_content:
    print("✓ Schools array correctly inserted")
if 'praiasFluviais' in html_content:
    print("✓ River beaches field present")

print("\n⚠️  Next steps:")
print("   1. Test in browser - DevTools Console: concelhos.length === 94, schools.length === 62")
print("   2. Test filters: 'non-hot + car + river beaches' should return interior areas")
print("   3. Verify Aljezur not in schools array")
print("   4. Test school proximity scoring")
